arches/app/search/components/map_filter.py

```diff
@@ -28,47 +30,57 @@ def append_dsl(self, search_results_object, permitted_nodegroups, include_provis
-                geoshape = GeoShape(
-                    field="geometries.geom.features.geometry", type=feature_geom["type"], coordinates=feature_geom["coordinates"]
-                )
-
-                invert_spatial_search = False
-                if "inverted" in feature_properties:
-                    invert_spatial_search = feature_properties["inverted"]
-
-                spatial_query = Bool()
-                if invert_spatial_search is True:
-                    spatial_query.must_not(geoshape)
-                else:
-                    spatial_query.filter(geoshape)
-
-                # get the nodegroup_ids that the user has permission to search
-                spatial_query.filter(Terms(field="geometries.nodegroup_id", terms=permitted_nodegroups))
 
-                if include_provisional is False:
-                    spatial_query.filter(Terms(field="geometries.provisional", terms=["false"]))
-
-                elif include_provisional == "only provisional":
-                    spatial_query.filter(Terms(field="geometries.provisional", terms=["true"]))
-
-                search_query.filter(Nested(path="geometries", query=spatial_query))
+                add_geoshape_query_to_search_query(feature_geom, feature_properties, permitted_nodegroups, include_provisional, search_query)
+
+        elif "featureid" in spatial_filter and "resourceid" in spatial_filter:
+            se = SearchEngineFactory().create()
+            main_query = Query(se)
+            nested_query = Nested(path="geometries")
+            match_feature = Match(field="geometries.geom.features.id", query=spatial_filter["featureid"])
+
+            # Create a Bool query for conditions inside the nested path
+            bool_nested_query = Bool()
+            bool_nested_query.must(match_feature.dsl)
+            nested_query.add_query(bool_nested_query.dsl)
+
+            bool_query = Bool()
+            match_resource = Term(field="resourceinstanceid", term=spatial_filter["resourceid"])
+            bool_query.must(match_resource.dsl)  # Match resource instance ID at the document level
+            bool_query.must(nested_query.dsl)  # Add the nested query
+            
+            # Set the entire bool query to the main query object
+            main_query.add_query(bool_query.dsl)
+
+            response = main_query.search(index=RESOURCES_INDEX)
+            geometries = []
+            for hit in response['hits']['hits']:
+                if len(geometries) > 0:
+                    break
+                for geom in hit['_source']['geometries']:
+                    if len(geometries) > 0:
+                        break
+                    for feature in geom['geom']['features']:
+                        if len(geometries) > 0:
+                            break
+                        if feature['id'] == spatial_filter["featureid"]:
+                            geometries.append(feature)
+
+            feature_geom = geometries[0]["geometry"]
```

Comment:
This throws an exception if the resource instance used as the source of the filter is deleted.

Reply:
Let me know if this is still giving you an error or exception after my latest push to branch